kart/base_dataset.py

```diff
@@ -105,6 +227,18 @@ def __init__(self, tree, path, repo, dirname=None):
 
         self.ensure_only_supported_capabilities()
 
+    @classmethod
+    def _init_meta_item_definitions(cls):
+        if hasattr(cls, "PATH_META_ITEMS"):
+            return
+        cls.PATH_META_ITEMS = {}
+        cls.PATTERN_META_ITEMS = []
+        for definition in cls.META_ITEMS:
+            if definition.path:
+                cls.PATH_META_ITEMS[definition.path] = definition
+            else:
+                cls.PATTERN_META_ITEMS.append(definition)
```

Comment:
super edge case, but this init pattern doesn't guard against a race condition where `cls.PATH_META_ITEMS` is set, an exception is thrown, the exception is caught and the dict/list are never populated.

Instead, populate a couple of local vars and then set them as class attributes at the end.

Also, maybe this should live in a metaclass so it doesn't have to happen during instance init

Reply:
Done